src/sentry/issues/auto_source_code_config/code_mapping.py

```diff
@@ -541,3 +570,27 @@ def find_roots(frame_filename: FrameInfo, source_path: str) -> tuple[str, str]:
     # validate_source_url should have ensured the file names match
     # so if we get here something went wrong and there is a bug
     raise UnexpectedPathException("Could not find common root from paths")
+
+
+# Based on # https://github.com/getsentry/symbolicator/blob/450f1d6a8c346405454505ed9ca87e08a6ff34b7/crates/symbolicator-proguard/src/symbolication.rs#L450-L485
+def get_path_from_module(module: str, abs_path: str) -> tuple[str, str]:
+    """This attempts to generate a modified module and a real path from a Java module name and filename.
+    Returns a tuple of (stack_root, source_path).
+    """
+    # An `abs_path` is valid if it contains a `.` and doesn't contain a `$`.
+    if "$" in abs_path or "." not in abs_path:
+        # Split the module at the first '$' character and take the part before it
+        # If there's no '$', use the entire module
+        file_path = module.split("$", 1)[0] if "$" in module else module
+        stack_root = module.rsplit(".", 1)[0].replace(".", "/")
+        return stack_root, file_path.replace(".", "/")
+
+    if "." not in module:
+        raise DoesNotFollowJavaPackageNamingConvention
+
+    # If module has a dot, take everything before the last dot
+    # com.example.foo.Bar$InnerClass -> com/example/foo/
```

Comment:
Looks like the code below doesn't have a trailing `/`. Maybe we should update the comment to avoid confusion?

Reply:
Good call; thanks!